pom.xml

```diff
@@ -302,6 +307,18 @@
                 </plugins>
             </build>
         </profile>
+        <profile>
+            <!-- Shading workaround for IDEs: resolve these modules from the local maven repo -->
```

Comment:
Mind expanding this with some instructions either here or in CONTRIBUTING.

Something along the lines of:
This project relocates some of its internal dependencies. Since IDEs can't resolve the relocated symbols on their own, a workaround is added: first `mvn install` the shaded module, then disable the `with-shaded` profile to force IDEs to resolve the relocated deps from the local repository

Reply:
Updated in both places